# E-Commerce Customer Intelligence Dashboard

## Notebook 03 – Payments Data Cleaning

### Objective

- Load payment dataset
- Inspect dataset
- Identify missing values
- Check duplicate records
- Validate data quality
- Clean dataset
- Save cleaned dataset

In [1]:
import pandas as pd

In [2]:
payments = pd.read_csv("../data/raw/olist_order_payments_dataset.csv")

In [3]:
payments.head()

,order_id,payment_sequential,payment_type,payment_installments,payment_value
0,b81ef226f3fe1789b1e8b2acac839d17,1,credit_card,8,99.33
1,a9810da82917af2d9aefd1278f1dcfa0,1,credit_card,1,24.39
2,25e8ea4e93396b6fa0d3dd708e76c1bd,1,credit_card,1,65.71
3,ba78997921bbcdc1373bb41e913ab953,1,credit_card,8,107.78
4,42fdf880ba16b47b59251dd489d4441a,1,credit_card,2,128.45


In [4]:
payments.shape

(103886, 5)

In [5]:
payments.columns

Index(['order_id', 'payment_sequential', 'payment_type',
       'payment_installments', 'payment_value'],
      dtype='object')

In [6]:
payments.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 103886 entries, 0 to 103885
Data columns (total 5 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   order_id              103886 non-null  object 
 1   payment_sequential    103886 non-null  int64  
 2   payment_type          103886 non-null  object 
 3   payment_installments  103886 non-null  int64  
 4   payment_value         103886 non-null  float64
dtypes: float64(1), int64(2), object(2)
memory usage: 4.0+ MB


## Dataset Inspection

### Observations

- The dataset contains 103,886 payment records.
- It has 5 columns.
- No datetime columns are present.
- All columns have appropriate data types.
- No datatype conversion is required.

In [7]:
payments.isnull().sum()

order_id                0
payment_sequential      0
payment_type            0
payment_installments    0
payment_value           0
dtype: int64

In [8]:
payments.duplicated().sum()

0

In [9]:
payments.duplicated(subset=["order_id", "payment_sequential"]).sum()

0

# Task Summary

## Payments Data Cleaning

### Actions Performed

- Loaded the Olist Payments dataset using Pandas.
- Inspected the dataset using `head()`, `shape`, `columns`, and `info()`.
- Verified that there are no missing values.
- Checked for duplicate rows.
- Verified that the combination of `order_id` and `payment_sequential` is unique.
- Confirmed that all data types are appropriate.
- No cleaning transformations were required.
- Saved the cleaned dataset.

### Output

`data/cleaned/payments_cleaned.csv`

In [10]:
payments.to_csv("../data/cleaned/payments_cleaned.csv", index=False)